### PR TITLE
Reorganizing the subprojects around

### PR DESCRIPTION
**Pull request by eed3si9n**
What used to be called "compile-integration" gets a better name "incrementalcompiler".
(I am thinking about moving `sbt.Compile` in there with better code-against-interface + sbt-datatype protection, eventually)

/review @jsuereth, @dwijnand, @Duhemm

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -62,34 +52,83 @@ def testedBaseSettings: Seq[Setting[_]] =
...
+    compilerInterfaceProj,
+    compileProj,
     compilerIvyProj).
   settings(
-    buildLevelSettings,
+    inThisBuild(Seq(
+      git.baseVersion := baseVersion,
+      bintrayPackage := "incrementalcompiler",
+      scmInfo := Some(ScmInfo(url("https://github.com/sbt/incrementalcompiler"), "git@github.com:sbt/incrementalcompiler.git")),
+      description := "Incremental compiler of Scala"
```
Comment (Duhemm) [inline build.sbt:71]: Shouldn't it say `Incremental compiler of sbt`?

Comment (eed3si9n) [inline build.sbt:71]: No I want this project to stand on its own.

File: build.sbt [hunk 2]
```diff
@@ -62,34 +52,83 @@ def testedBaseSettings: Seq[Setting[_]] =
...
+    compilerBridgeProj,
+    compilerApiInfoProj,
     classpathProj,
     classfileProj,
-    compileInterfaceProj,
-    compileIncrementalProj,
-    compilePersistProj,
-    compilerProj,
-    compilerIntegrationProj,
+    compilerInterfaceProj,
```
Comment (Duhemm) [inline build.sbt]: `compilerInterfaceProj` is there twice

**Comment by Duhemm** [comment]
Apart from one really minor thing, LGTM

**Pull request merged**